src/widgets/table/table.rs

```diff
@@ -1484,7 +1364,264 @@ mod tests {
+                Some(0),
+                Buffer::with_lines(vec![
+                    "hello  world   ", // row 1
+                    "               ", // row 2
+                    "               ", // row 3
+                ]),
+            );
+            // highlight_symbol rendered because selection is made
+            test(
+                HighlightSpacing::WhenSelected,
+                Some(0),
+                Buffer::with_lines(vec![
+                    ">>>hello  world", // row 1
+                    "               ", // row 2
+                    "               ", // row 3
+                ]),
+            );
+            // no highlight_symbol rendered because no selection is made
+            test(
+                HighlightSpacing::Always,
+                Some(0),
+                Buffer::with_lines(vec![
+                    ">>>hello  world", // row 1
+                    "               ", // row 2
+                    "               ", // row 3
+                ]),
+            );
+        }
+
+        #[test]
+        fn spacers_priority_constrained_with_column_spacing() {
+            fn test(
+                highlight_spacing: HighlightSpacing,
+                selection: Option<usize>,
+                expected: Buffer,
+            ) {
+                let table = Table::default()
+                    .rows(vec![Row::new(vec!["hello", "12345"])])
+                    .highlight_spacing(highlight_spacing)
+                    .highlight_symbol(">>>")
+                    .column_spacing(1);
+                let area = Rect::new(0, 0, 10, 3);
+                let mut buf = Buffer::empty(area);
+                if let Some(s) = selection {
+                    let mut state = TableState::default().with_selected(Some(s));
+                    StatefulWidget::render(table, area, &mut buf, &mut state);
+                    assert_buffer_eq!(buf, expected);
+                } else {
+                    Widget::render(table, area, &mut buf);
+                    assert_buffer_eq!(buf, expected);
+                }
+            }
+
+            // column spacing is prioritized over everything constraint
+            test(
+                HighlightSpacing::Never,
+                None,
+                Buffer::with_lines(vec![
+                    "hello 1234", // row 1
+                    "          ", // row 2
+                    "          ", // row 3
+                ]),
+            );
+            test(
+                HighlightSpacing::WhenSelected,
+                None,
+                Buffer::with_lines(vec![
+                    "hello 1234", // row 1
+                    "          ", // row 2
+                    "          ", // row 3
+                ]),
+            );
+            // highlight_symbol space always allocated
+            // because column widths is calculated after
+            // when column widths are calculated, column spacing is prioritized
```

Comment:
This comment requires some implied understanding of the implementation. Taking a step back from this, how would you explain it if someone didn't know about the implementation detail.

Th comment is difficult to parse in a way that makes grammatical sense "highlight_symbol space always allocated because _column widths is calculated after when column widths are calculated_, column spacing is prioritized". I'm assuming those are three sentences perhaps?

What is the actual property that is being tested for correctness with this test? highlight symbol or column spacing priority? (It's not clear)

Reply:
good catch, this comment was garbled. I've updated it with something that is easier to parse. Also added more tests to describe what is going on.